pkg/model/network.go

```diff
@@ -148,35 +151,74 @@ func (b *NetworkModelBuilder) Build(c *fi.ModelBuilderContext) error {
 		}
 	}
 
-	for _, zone := range privateZones.List() {
+	for i, zone := range privateZones.List() {
+
 		utilitySubnet, err := b.LinkToUtilitySubnetInZone(zone)
 		if err != nil {
 			return err
 		}
 
+		// Has an existing NgwId been entered?
+		// NGWs look like this: ngwId: nat-09c4180b76a36ca2c
+		ngwId := b.Cluster.Spec.Subnets[i].NgwId
```

Comment:
I don't think that the subnets order need match the zone order.

Also this shows up a deeper issue: we're configuring NAT gateways per subnet, but we really should be setting it per zone (route table).  And if it turns out we could share a route table (#1368) it is even more complicated :-(